internal/uvm/create_lcow.go

```diff
@@ -204,6 +206,17 @@ func CreateLCOW(ctx context.Context, opts *OptionsLCOW) (_ *UtilityVM, err error
 		return nil, fmt.Errorf("KernelDirectBoot is not support on builds older than 18286")
 	}
 
+	memory := &hcsschema.Memory2{
+		SizeInMB:             memorySizeInMB,
+		AllowOvercommit:      opts.AllowOvercommit,
+		EnableDeferredCommit: opts.EnableDeferredCommit,
+	}
+	if opts.EnableColdDiscardHint && osversion.Get().Build < 18967 {
```

Comment:
You can just do this validation but always assign EnableColdDiscardHints: opts.EnableColdDiscardHint in the struct.

Reply:
I believe that will cause a failure on the hcs side if the windows build does not support cold discard hints since the hcs doesn't know how to handle that field in the json object.

Reply:
The field is marked `omitempty` in json right? So when its false it wont be marshalled and you verify that when it is true the build needs to be >= 18967 right?

Reply:
Right, fixed.